pkg/iam/iam.go

```diff
@@ -0,0 +1,153 @@
+package iam
+
+import (
+	"github.com/urfave/cli"
+
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sts"
+)
+
+func RegisterCommands(app *cli.App) {
+
+	app.Commands = append(app.Commands, cli.Command{
+		Name:		"iam",
+		Usage:		"AWS IAM",
+		Category:	"AWS",
+		Subcommands: []cli.Command{
+			{
+				Name: "whoami",
+				Action: func(c *cli.Context) error {
+					name, err := getIAMUserViaGetCallerIdentity()
+					if err == nil {
+						fmt.Println(name)
+					}
+					return err
+				},
+			},
+			{
+                Name: "ar",
+                Action: func(c *cli.Context) error {
+					role := c.String("role")
+					creds, err := AssumeRoleCredentials(role)
+					if err != nil {
+						return err
+					}
+					fmt.Println(creds)
+					return nil
+                },
+				Flags: []cli.Flag{
+					cli.StringFlag{
+						Name: "role",
+						Usage: "the name of the role to assume",
+					},
+				},
+			},
+		},
+	})
+}
+
+func NewSTSService(sc *sts.Credentials) (*sts.STS, error) {
+	sess, err := session.NewSessionWithOptions(session.Options{
+		Config: aws.Config{
+			Credentials: credentials.NewStaticCredentials(*sc.AccessKeyId, *sc.SecretAccessKey, *sc.SessionToken),
+		},
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return sts.New(sess), nil
+}
+
+func AssumeRoleCredentials(role string) (*sts.Credentials, error) {
+	name, _ := getIAMUserViaGetCallerIdentity()
+
+	params := &sts.AssumeRoleInput{
+		RoleArn: aws.String(fmt.Sprintf("arn:aws:iam::590710864528:role/%s", role)),
```

Comment:
How much of these domain specific things could be shifted to a config or env vars? Hardcoding this means this tool cannot be used outside of the organization. It also means a change to this means you have to recompile and redistribute the binary to everyone in the org.

Reply:
Agreed here, we can read this out of ~/.aws/config can't we?